### DISCUSSION TITLE
Heads up: the Sass legacy JS API will start emitting warnings soon

### DISCUSSION THREAD

**Discussion by nex3** [vercel/next.js#67931] [Ideas]
### Goals

Sorry for the template abuse, but I'm filing this on behalf of the Sass team as a heads-up that we're planning to release a version of Sass that begins to emit deprecation warnings for uses of [the legacy JS API](https://sass-lang.com/documentation/js-api/#md:legacy-api). Since this is currently the default for Next.js, it might be a good idea to move to using the [modern API](https://sass-lang.com/documentation/js-api/#md:usage) as the default instead.

### Background

N/A

### Proposal

N/A

**Comment by iamlinkus** [comment]
I'm extremely interested on this getting fixed. Is there a way to force next.js to use the modern api as default as of right now? It doesn't look like we have the ability to set it using`sassOptions`.

**Reply by mrienstra** [reply]
This is the issue to follow: https://github.com/vercel/next.js/issues/71638 (already mentioned by @ariesclark)

**Comment by ariesclark** [comment]
For those looking to silence these errors until Next.js resolves this issue internally.
```ts
const nextConfig: NextConfig = {
  sassOptions: {
    silenceDeprecations: ["legacy-js-api"],
  }
}
```
https://nextjs.org/docs/app/building-your-application/styling/sass and https://sass-lang.com/documentation/breaking-changes/legacy-js-api/#silencing-warnings

Credit to https://github.com/vercel/next.js/issues/71638#issuecomment-2431137842.

**Reply by CaptainN** [reply]
This does absolutely nothing.

**Reply by CaptainN** [reply]
I have to assume that's to be expected since it's not mentioned anywhere in the docs?

**Reply by chrimesdev** [reply]
This does nothing for me either

**Comment by Netail** [comment]
@devjiwonchoi What happened to https://github.com/vercel/next.js/pull/72423 (sass-loader bump to 16.x), is there something else coming?

Sorry had to tag you here as the sass PRs and GitHub issues are already locked

**Reply by devjiwonchoi** [reply]
Hi, no worries, thanks for flagging! I've investigated and noticed Next.js doesn't directly use the Sass APIs, so it seems not urgent to bump the sass-loader version, silencing the warning for now. Will eventually upgrade - please correct me if I'm wrong!

**Reply by Netail** [reply]
I mean it's not an issue atm, but at some point they will deprecated the legacy API in sass itself, meaning the sass-loader won't work with the latest sass versions out of the box anymore (only by switching the `api` option in the sassOptions to "modern", which could leave a lot of users confused, unknown of this option). So it's more to prevent this period where users are confused why the latest sass version won't work out of the box anymore.

The new sass-loader versions uses the modern API by default, but does differ a bit in sassOptions. If the sass-loader version is bumped earlier, we can prevent this issue, but does involve a breaking change for users who use the legacy API options. (However, they could use the "legacy" option for `api`)

**Reply by iamlinkus** [reply]
I think exposing the choice for api (modern/legacy) would be tha sanest way to go here. I agree that not addressing this now might cause lots of issues unpredictably soon.

**Reply by devjiwonchoi** [reply]
Makes sense, will follow up soon. Thank you all! 🙏

**Reply by Netail** [reply]
@devjiwonchoi I assume the next release (besides patches) will be 16.x, do you think it's time to bump sass-loader to 16.0.5 + removing the silenceDeprecations? https://github.com/vercel/next.js/pull/83091 Not sure how breaking changes are tracked for the dev blog

**Comment by alexsamofalov** [comment]
`const nextConfig: NextConfig = {
    sassOptions: {
        quietDeps: true
    }
};`

**Comment by lipis** [comment]
Still emitting bunch of warnings and nothing from the `sassOptions` that I tried doesn't work.

**Reply by stephenhmarsh** [reply]
I have the appropriate `sassOptions` config to silence the warnings, and it works with Next 15.3.1 until I use Storybook.  However Storybook is bundling my components is not respecting Next's `sassOptions`.

**Comment by Netail** [comment]
sass-loader will be bumped to v16 with Next.js v16 :)
https://github.com/vercel/next.js/commit/0f781a6a41a7823f7236ac34aedbabddda8a7cfd
